org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServerWrapper.java

```diff
@@ -451,14 +454,14 @@ public synchronized void stop() {
 				}
 			}
 
-			if (serverFuture != null) {
-				serverFuture.cancel(true);
-			}
-
 			if (languageServerInstance != null) {
 				languageServerInstance.exit();
 			}
 
+			if (serverFuture != null) {
```

Comment:
I move this serverFuture cancel to avoid having Piped stream closing error when exit() is called.

Reply:
I have tested the change and I see the same exception patterns with and without when running the tests.

Reply:
@rubenporras see [my bug report](https://github.com/eclipse/lsp4e/issues/681) - on Eclipse shutdown this code is not executed, but results in same error, making reproduction attempts confusing.